readme.md

```diff
@@ -1,111 +1,109 @@
+
+### Environments and Data
 
-- the app's credentials - client id, client secret
-- the app's callback url
-- the version number of the API
-- the app's environment (web location where the app is registered)
-- The FHIR call retry settings
+The Blue Button 2.0 API is available in V1 and V2 in a Sandbox and Production environment.
 
-the configuration is in json format and stored in a local file, the default location
-is current working directory with file name: .bluebutton-config.json
+- Sandbox location: https://sandbox.bluebutton.cms.gov
+- Production location: https://api.bluebutton.cms.gov
 
-A sample configuration json:
+Version data formats:
+
+- V1: FHIR STU2
+- V2: FHIR R4
+
+Sample configuration JSON with default version and environment:
 
 ```
 {
   "clientId": "foo",
   "clientSecret": "bar",
   "callbackUrl": "https://www.fake.com/",
-  "retrySettings": {
-    "total": 3,
-    "backoffFactor": 5,
-    "statusForcelist": [500, 502, 503, 504, 508]
-  }
 }
 
 ```
 
-OAUTH2.0 parameters: clientId, clientSecret, callbackUrl, they are mandatory and must be provided
-
-| parameter    | value                   | Comments                        |
-| ------------ | ----------------------- | ------------------------------- |
-| clientId     | "foo"                   | oauth2 client id of the app     |
-| clientSecret | "bar"                   | oauth2 client secret of the app |
-| callbackUrl  | "https://www.fake.com/" | oauth2 callback url of the app  |
-
-For application registration and client id and client secret, please refer to:
-[Blue Button 2.0 API Docs - Try the API](https://bluebutton.cms.gov/developers/#try-the-api)
+If needed, you can set your application's target environment and API version.
 
-FHIR requests retry:
+Example:
 
-Retry is enabled by default for FHIR requests, retry_settings: parameters for exponential back off retry algorithm
-
-| retry parameter  | value (default)      | Comments                         |
-| ---------------- | -------------------- | -------------------------------- |
-| backoff_factor   | 5                    | back off factor in seconds       |
-| total            | 3                    | max retries                      |
-| status_forcelist | [500, 502, 503, 504] | error response codes to retry on |
-
-the exponential back off factor (in seconds) is used to calculate interval between retries by below formular, where i starts from 0:
-
-backoff factor \* (2 \*\* (i - 1))
+```
+{
+  "clientId": "foo",
+  "clientSecret": "bar",
+  "callbackUrl": "https://www.fake.com/",
+  "version": "2",
+  "environment": "PRODUCTION"
+}
 
-e.g. for backoff_factor is 5 seconds, it will generate wait intervals: 2.5, 5, 10, ...
+```
 
-to disable the retry: set total = 0
+## Usage <a name="usage"></a>
 
-## Sample Usages: Obtain Access Grant, Probe Scope, and Access Data <a name="usages"></a>
+The following pseudocode shows the SDK used with a Node JS Express server. This code walks through:
```

Comment:
```suggestion
The following example code shows the SDK used with a Node JS Express server. This code walks through:
```

I think pseudocode should be replaced here, since these are actual coding examples.

Reply:
I debated about this as well.  In one of my drafts I removed it, but it does require some setup for the code to run.  But I agree,  we can probably change this as Dave shows above.